python/dgl/distgnn/partition/libra_partition.py

```diff
@@ -0,0 +1,577 @@
+        adj        = th.zeros([num_nodes, nc - 1], dtype=th.int32)
+        inner_node = th.zeros(num_nodes, dtype=th.int32)
+        lf         = th.zeros(num_nodes, dtype=th.int32)
+        ldt = ldt_ar[0]
+
+        try:
+            feat = g_orig.ndata['feat']
+        except:
+            feat = g_orig.ndata['features']
+
+        try:
+            labels = g_orig.ndata['label']
+        except:
+            labels = g_orig.ndata['labels']
+
+        trainm = g_orig.ndata['train_mask']
+        testm = g_orig.ndata['test_mask']
+        valm = g_orig.ndata['val_mask']
+
+        feat_size = feat.shape[1]
+        gfeat = th.zeros([num_nodes, feat_size], dtype=feat.dtype)
+
+        glabels = th.zeros(num_nodes, dtype=labels.dtype)
+        gtrainm = th.zeros(num_nodes, dtype=trainm.dtype)
+        gtestm = th.zeros(num_nodes, dtype=testm.dtype)
+        gvalm = th.zeros(num_nodes, dtype=valm.dtype)
+
+        ## build remote node databse per local node
+        ## gather feats, train, test, val, and labels for each partition
+        libra2dgl_build_adjlist(feat, gfeat, adj, inner_node, ldt, gdt_key,
+                                gdt_value, node_map, lf, lftensor, num_nodes,
+                                nc, i, feat_size, labels, trainm, testm, valm,
+                                glabels, gtrainm, gtestm, gvalm, feat.shape[0])
+        
+
+        g.ndata['adj'] = adj   ## databse of remote clones
+        g.ndata['inner_node'] = inner_node  ## split node '0' else '1'
+        g.ndata['feat'] = gfeat    ## gathered features
+        g.ndata['lf'] = lf  ## 1-level tree among split nodes
+
+        g.ndata['label'] = glabels
+        g.ndata['train_mask'] = gtrainm
+        g.ndata['test_mask'] = gtestm
+        g.ndata['val_mask'] = gvalm
+
+        lf = g.ndata['lf']
+        print("Writing partition {} to file".format(i), flush=True)
+        
+        part = g
+        part_id = i
+        part_dir = os.path.join(out_path, "part" + str(part_id))
+        node_feat_file = os.path.join(part_dir, "node_feat.dgl")
+        edge_feat_file = os.path.join(part_dir, "edge_feat.dgl")
+        part_graph_file = os.path.join(part_dir, "graph.dgl")
+        part_metadata['part-{}'.format(part_id)] = {'node_feats': node_feat_file,
+                                                    'edge_feats': edge_feat_file,
+                                                    'part_graph': part_graph_file}
+        os.makedirs(part_dir, mode=0o775, exist_ok=True)
+        save_tensors(node_feat_file, part.ndata)
+        save_graphs(part_graph_file, [part])
+        
+        del g
+        del gg[0]
+        del ldt
+        del ldt_ar[0]
+        
+    with open('{}/{}.json'.format(out_path, graph_name), 'w') as outfile:
+        json.dump(part_metadata, outfile, sort_keys=True, indent=4)
+        
+    return gg, node_map
+
+
+## \brief Driver function for Libra output to DGL graph format conversion
+def run_libra2dgl(dataset, resultdir, nc):
```

Comment:
This seems to be converting Libra partition directory results to DistGNN configurations.  Does it have to contain the dataset argument?

I imagine this file will be exposing Python functions for end users to partition their graphs, so I guess ideally it will only have one function `vertex_cut_partition` (or `libra_partition` or whatever) that takes in a single DGLGraph and generates DistGNN configurations.  Is there any concern to combine this function together with `vertex_cut_partition`?

All other functions can be removed or moved to a script file in `tools`.